working-docs/transient.md

```diff
@@ -0,0 +1,379 @@
+
+  - Although BCB can affect conditional branches that do serve as
+    “security-critical protection mechanisms” (for example, in
+    software sandboxes), BCB can also affect branches that serve a
+    functional purpose, such as dynamic type checking \[2\]. Therefore
+    CWE-1037 is too narrow.
+
+  - The conditional branch instructions affected by BCB are not
+    “hardware logic for error handling and security checks.” Hence
+    CWE-1264 does not apply.
+
+  - BCB can be exploited over a network \[3\] without shared hardware
+    resources, and therefore CWE-1303 does not apply.
+
+  - CWE-1342 implies that the processor should be clearing
+    microarchitectural state, which is impractical or infeasible for
+    many vulnerabilities, including BCB.
+
+The objective of this document is to define
+**<span class="underline">four</span>** new CWEs that address the root
+causes of transient execution attacks, and with language that is
+sufficiently generic to benefit the entire CWE community.
+
+New CWE Proposals
+--------------------------------
+
+### CWE-A: Processor Event Causes Transient Execution
+
+#### Description
+
+A processor event may allow subsequent operations to execute transiently
+(the operations execute without committing to architectural state).
+
+#### Extended Description
+
+Many commodity processors execute operations (such as instructions,
+etc.) out-of-order. For example, a processor may fetch the sequence of
+instructions `A;B;C`, but then execute them in a different order, such as
+`A;C;B`. This out-of-order execution sequence may produce the same result
+as an in-order sequence if the input to `C` does not depend on the output
+of `B`. Although operations on these processors may execute out-of-order,
+they commit in-order to the processor’s architectural state. Hence, all
+operations appear to software as though they had been executed in-order
+by the processor.
+
+Some processor events can cause an out-of-order processor to flush its
+pipeline, discarding the results of operations that have already
+executed but not committed to the processor’s architectural state. For
+example, suppose that while executing `B` in the out-of-order sequence
+`A;C;B`, the processor encounters an event that causes a pipeline flush.
+If `C` executed before `B`, the event may require `C`’s results to be
+discarded during the flush.
+
+When operations execute but do not commit to the processor’s
+architectural state, this is commonly referred to as *transient
+execution*. Operations that execute transiently may have
+microarchitectural side effects that can be detected using timing or
+power analysis techniques. These techniques may allow an attacker to
+infer information about the operations that executed transiently. For
+example, the attacker may be able to infer program data that was
+accessed or used by those operations.
+
+### CWE-B: Transient Data Forwarding from an Operation that Triggers a Processor Event
```

Comment:
Gananand: Based on the description, perhaps this should be “Incorrect forwarding of transient data”? Or “Incorrect forwarding of transient data that is observable after an architectural state commit”?